src/components/DuetRoom.tsx

```diff
@@ -114,6 +119,23 @@ const DuetRoom: React.FC<{ maybeRoomId: string | null; isCreate: boolean }> = ({
     }, 1000);
   }, [timeToStart]);
 
+  useEffect(() => {
+    if (piece === undefined) return;
+
+    async function fetchSongMIDI() {
+      try {
+        const song = await songsAPI.getSongWithContent(piece!);
```

Comment:
Is this force-unwrap needed?

Reply:
TS complains :(

Reply:
Oh maybe because it's async. What if we move `if (piece === undefined) return;` into the function?

Reply:
yeah that works, but wouldn't it be better to just return at the start of the callback, instead of calling fetchSongMIDI then returning?

Reply:
I don't know js enough :/

I think the reason why TypeScript thinks `piece` might be empty inside the async function despite the early return before the function declaration is because the async function might not be executed immediately and there is a slight chance that after the check and before the async function starts executing the piece becomes empty.

If we conduct the check within the function, then it must be present when we use it, because there is no 'context switch' once a function starts executing (except for when interrupted by `await`)

Reply:
ah that's true, ill just put both then